bin/strata-client/src/args.rs

```diff
@@ -1,171 +1,222 @@
 use std::path::PathBuf;
 
+use anyhow::anyhow;
 use argh::FromArgs;
-use bitcoin::Network;
-use strata_config::{
-    bridge::RelayerConfig, btcio::BtcioConfig, BitcoindConfig, ClientConfig, ClientMode, Config,
-    ExecConfig, FullNodeConfig, RethELConfig, SequencerConfig, SyncConfig,
-};
+use serde::de::DeserializeOwned;
+use serde_json::{from_value, to_value, Value};
+use strata_config::Config;
+
+/// Configs overridable by environment. Mostly for sensitive data.
+#[derive(Debug, Clone)]
+pub struct EnvArgs {
+    // TODO: relevant items that will be populated from env vars
+}
+
+impl EnvArgs {
+    pub fn from_env() -> Self {
+        // Here we load particular env vars that should probably override the config.
+        Self {}
+    }
+
+    /// Override some of the config params from env.
+    pub fn override_config(&self, _config: &mut Config) -> bool {
+        // Override attributes
+        true
+    }
+}
 
 #[derive(Debug, Clone, FromArgs)]
-#[argh(description = "Alpen Strata sequencer")]
+#[argh(description = "Strata client")]
 pub struct Args {
-    // TODO: default config location
+    // Config non-overriding args
     #[argh(option, short = 'c', description = "path to configuration")]
-    pub config: Option<PathBuf>,
+    pub config: PathBuf,
 
+    // Config overriding args
     #[argh(
         option,
         short = 'd',
         description = "datadir path that will contain databases"
     )]
     pub datadir: Option<PathBuf>,
 
-    #[argh(option, short = 'h', description = "JSON-RPC host")]
-    pub rpc_host: Option<String>,
+    #[argh(switch, description = "is sequencer")]
+    pub sequencer: bool,
 
-    #[argh(option, short = 'r', description = "JSON-RPC port")]
-    pub rpc_port: Option<u16>,
+    #[argh(option, description = "rollup params")]
+    pub rollup_params: Option<PathBuf>,
 
-    #[argh(option, description = "bitcoind RPC host")]
-    pub bitcoind_host: Option<String>,
+    #[argh(option, description = "rpc host")]
+    pub rpc_host: Option<String>,
 
-    #[argh(option, description = "bitcoind RPC user")]
-    pub bitcoind_user: Option<String>,
+    #[argh(option, description = "rpc port")]
+    pub rpc_port: Option<u16>,
 
-    #[argh(option, description = "bitcoind RPC password")]
-    pub bitcoind_password: Option<String>,
+    #[argh(option, short = 'o', description = "generic config overrides")]
+    pub overrides: Vec<String>,
+}
 
-    /// Max retries for Bitcoin RPC calls.
-    #[argh(option, description = "max retries for bitcoin RPC (default: 3)")]
-    pub bitcoind_retry_count: Option<u8>,
+impl Args {
+    /// Overrides config. First overrides with the generic overrides passed via -o and then
+    /// overrides the result with a couple of commonly passed args.
```

Comment:
I don't see "a couple" here. This is not that informative. This means that it will override by chunks of pairs of configurations (the "couple")?
Maybe do `s/a couple/some`.